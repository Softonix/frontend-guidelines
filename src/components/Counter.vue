<template>
  <div class="my-2" data-testid="root-wrapper">
    <h2><b>{{ title }}</b></h2>
    <p>Times clicked: {{ count }}</p>
    <el-button @click="increment">increment</el-button>
  </div>
</template>

<script setup lang="ts">
withDefaults(defineProps<{
  title: string
}>(), {
  title: 'This is a demo for unit testing'
})

const count = ref(0)

function increment () {
  count.value += 1
}
</script>
