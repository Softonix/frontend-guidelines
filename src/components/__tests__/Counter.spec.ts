import { it, expect } from 'vitest'
import { render, fireEvent } from '@testing-library/vue'
import Counter from '../Counter.vue'

it('should increments value on click', async () => {
  const { getByText } = render(Counter)

  // screen has all queries that you can use in your tests.
  // getByText returns the first matching node for the provided text, and
  // throws an error if no elements match or if more than one match is found.
  getByText('Times clicked: 0')

  const button = getByText('increment')

  // Dispatch a native click event to our button element.
  await fireEvent.click(button)
  await fireEvent.click(button)

  getByText('Times clicked: 2')
})

it('should renders the title prop correctly', () => {
  const title = 'Test Title'

  const { html } = render(Counter, {
    props: {
      title
    }
  })

  // Assert that the title is rendered correctly
  expect(html()).toContain(title)
})

it('should renders the default title if prop does not provided', () => {
  const defaultTitle = 'This is a demo for unit testing'

  const { html } = render(Counter, {
    props: {
      title: defaultTitle
    }
  })

  // Assert that the title is rendered correctly
  expect(html()).toContain(defaultTitle)
})
