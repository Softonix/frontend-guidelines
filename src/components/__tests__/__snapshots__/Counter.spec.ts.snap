// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should create a Counter component snapshot 1`] = `
<div
  class="my-2"
  data-testid="root-wrapper"
>
  <h2>
    <b>
      This is a demo for unit testing
    </b>
  </h2>
  <p>
    Times clicked: 0
  </p>
  <button
    ariadisabled="false"
    class="el-button"
    type="button"
  >
    <!--v-if-->
    <span
      class=""
    >
      
      increment
      
    </span>
  </button>
</div>
`;
